samcli/lib/providers/sam_layer_provider.py

```diff
@@ -55,11 +63,11 @@ def get(self, name):
             raise ValueError("Layer name is required")
 
         for layer in self._layers:
-            if layer.name == name:
+            if posixpath.join(layer.stack_path, layer.name) == name or layer.name == name:
```

Comment:
Can we add some description about why we are comparing name with path of the layer?

Reply:
Added!